app/handlers/handlers_groups.py

```diff
@@ -0,0 +1,71 @@
+from datetime import timedelta
+
+import kopf
+
+from app.api import TwingateAPIClient
+from app.api.exceptions import GraphQLMutationError
+from app.crds import TwingateGroupCRD
+from app.handlers import fail, success
+
+
+@kopf.on.resume("twingategroup")
+@kopf.on.create("twingategroup")
+@kopf.on.update("twingategroup")
+def twingate_group_create_update(body, spec, logger, memo, patch, **kwargs):
+    logger.info("twingate_group_reconciler: %s", spec)
+    settings = memo.twingate_settings
+    client = TwingateAPIClient(settings)
+
+    if diff := kwargs.get("diff"):
+        logger.info("Diff: %s", diff)
+        # If ID changed from `None to value we just created it no need to update
+        # diff is `(('add', ('spec', 'id'), None, 'R3JvdXA6MjAxNjc4OQ=='),)`
+        if (
+            len(diff) == 1
+            and diff[0][0] == "add"
+            and diff[0][1] == ("spec", "id")
+            and diff[0][2] is None
+        ):
+            return success()
+
+    crd = TwingateGroupCRD(**body)
+    group_id = crd.spec.id
+    if group_id:
+        logger.info("Updating group with name='%s'", crd.spec.name)
+        try:
+            client.group_update(crd.spec)
+        except GraphQLMutationError as gqlerr:
+            logger.error("Failed to update group: %s", gqlerr)
+            if "does not exist" in gqlerr.message:
+                patch.spec["id"] = None
+            return fail(error=gqlerr.message)
```

Comment:
I wonder if we should do this in the client and throw some custom exception like `NotFound`. We could do this for other entities too?

Reply:
Maybe...  but if we do its its own PR - design change for client API.

Reply:
Added issue #400